Assets/Scripts/Battlefield/CreatureScripts/UniqueCreature.cs

```diff
@@ -24,7 +24,16 @@ public class UniqueCreature : UnitBase
         [HideInInspector]
         BrainManager brain;
         public bool isEnemy;
-        
+        [HideInInspector]
+        public GridHolder gridHolder;
+        bool start = true;
+        bool onMoveTile = false;
+        LineRenderer lr;
+
+        //Used for astar
+        private List<Tile> path;
+        private int tileOnPath = 0;
```

Comment:
Should be part of a MovementSystem abstraction such as an object that takes and agent, startPoint and Destination and does AStar and moves your unit. This is unnecessary and complicates the code to someone who is expecting only the data for a unit, not the internals of what algorithm it uses to move.